test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,38 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), borrow_loaned_message) {
+  rmw_publisher_options_t options = rmw_get_default_publisher_options();
+  constexpr char topic_name[] = "/test";
+  const rosidl_message_type_support_t * ts =
+    ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BasicTypes);
+  rmw_publisher_t * pub =
+    rmw_create_publisher(node, ts, topic_name, &rmw_qos_profile_default, &options);
+  ASSERT_NE(nullptr, pub) << rmw_get_error_string().str;
+  void * msg_pointer{nullptr};
+  rmw_ret_t ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
+  if (ret != RMW_RET_UNSUPPORTED)
+  {
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
```

Comment:
@Lobotuerk if borrowing is successful, the message has to be explicitly returned or we'll leak.

Reply:
By explicitly returning, do you mean with ```rmw_return_loaned_message_from_publisher```?